crates/services/p2p/src/p2p_service.rs

```diff
@@ -1275,11 +1327,33 @@ mod tests {
                         let broadcast_request = broadcast_request.clone();
                         matches!(node_b.publish_message(broadcast_request), Err(PublishError::Duplicate));
 
-                        break
+                        match acceptance {
+                            GossipsubMessageAcceptance::Reject | GossipsubMessageAcceptance::Ignore => {
+                                break
```

Comment:
Seems we expect panic here

Reply:
Maybe we shouldn't even break here? If we want to test that node c doesn't receive ignored messages then we should leave node b operating long enough for it to potentially send a message it shouldn't have.

The tricky part would be that we'd need some kind of timeout to determine when it's safe to assume node b hasn't sent a bad message and we can end the test 🤔

Reply:
I'm with @Voxelot but yeah deciding when to exit is a bit tricky, can think of something maybe with timeouts, or sending another message...

Reply:
> Seems we expect panic here

I don't get it, why panic, the test is a success at that point (if we're looking at the same thing)

Reply:
> I don't get it, why panic, the test is a success at that point (if we're looking at the same thing)

The comment below says "// the `exit` should happen in Node C". So I assume that `break` in the node C should end the test=)

Reply:
yes that is correct, but that's just a comment to describe the flow of the test, it should not panic when it hits the line of the comment, cuz it needs to hit that line, ie the message validity is in `Accept` state and that is fine

Reply:
But why do we need two `break`?

Reply:
depends on type of test, you can pass Ignore/Reject or Accept case, depending which one happens, the test should either exit in Node B or Node C.

That's just a reusable function  for tests like
```
#[tokio::test]
    #[instrument]
    async fn gossipsub_broadcast_block_with_accept() {
        gossipsub_broadcast(
            GossipsubBroadcastRequest::NewBlock(Arc::new(Block::default())),
            GossipsubMessageAcceptance::Accept,
        )
        .await;
    }

    #[tokio::test]
    #[instrument]
    async fn gossipsub_broadcast_block_with_ignore() {
        gossipsub_broadcast(
            GossipsubBroadcastRequest::NewBlock(Arc::new(Block::default())),
            GossipsubMessageAcceptance::Ignore,
        )
        .await;
    }
```

Reply:
lets defer the timeout issue to a followup tech-debt item